test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs

```diff
@@ -38,19 +39,38 @@ public static DiagnosticResult Diagnostic (DiagnosticDescriptor descriptor)
 			(string, string)[]? globalAnalyzerOptions = null)
 			=> CreateCompilation (CSharpSyntaxTree.ParseText (src), globalAnalyzerOptions);
 
+		static List<MetadataReference> GetLatestNETCoreReferenceAssemblies ()
+		{
+			FileInfo fiSPCL = new FileInfo (typeof (int).Assembly.Location);
```

Comment:
Do we really need to filter and check everything?
Why not to give the metadata references without modifying them, like:
```
var referenceSPC = MetadataReference.CreateFromFile (typeof (object).Assembly.Location);
if (referenceSPC is null)
  throw new FileNotFoundException ("Could not find System.Private.CoreLib.dll");
var comp = CSharpCompilation.Create (
  assemblyName: Guid.NewGuid ().ToString ("N"),
	syntaxTrees: new SyntaxTree[] { src },
	references: new[] { referenceSPC },
	new CSharpCompilationOptions (OutputKind.DynamicallyLinkedLibrary));
```

Reply:
That would only add a reference to `System.Private.CoreLib`; it'd probably work for the tests that we currently have, but if someone were to add more complex tests cases, say having something like:

```C#
List<Type> types = new List<Type> { typeof(MyType) };
```

it'd fail... also, this doesn't seem to take much time, so I'd rather keep it this way.

Reply:
Wrote too fast 🙃. Everything seems to work fine even when just referencing SPCL, so I took your advice. I'll let @agocke comment on this.